src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -231,15 +242,9 @@ static void F(C<object>? a, C<object?> b)
                 // (8,32): warning CS8619: Nullability of reference types in value of type 'C<object?>' doesn't match target type 'C<object>'.
                 //             var c = new[] { a, b };
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b").WithArguments("C<object?>", "C<object>").WithLocation(8, 32),
-                // (9,13): warning CS8602: Possible dereference of a null reference.
-                //             c[0].ToString();
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c[0]").WithLocation(9, 13),
```

Comment:
>Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c[0]").WithLocation(9, 13), [](start = 16, length = 76)

Removal of this warning feels unexpected. It looks like a is definitely null.

Reply:
Added comment.

---
In reply to: [169863416](https://github.com/dotnet/roslyn/pull/24935#discussion_r169863416) [](ancestors = 169863416)